### PR TITLE
Added Connection Troubleshooting guide

### PR DESCRIPTION
**Pull request by DBirtolo-mdb**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-29489>
Staging - <https://docs-mongodbcom-staging.corp.mongodb.com/golang/docsworker-xlarge/DOCSP-29489/connection-troubleshooting/#std-label-golang-connection-troubleshooting>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?

### CODE REVIEW

File: source/connection-troubleshooting.txt [hunk 1]
```diff
@@ -0,0 +1,242 @@
+.. _golang-connection-troubleshooting:
+
+================================
+Connection Troubleshooting Guide
```
Comment (rustagir) [inline source/connection-troubleshooting.txt:4]: Q: could we just call this connection troubleshooting? many pages are "guides" but we omit that word from the titles

File: source/connection-troubleshooting.txt [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+================================
+
+.. default-domain:: mongodb
```
Comment (rustagir) [inline source/connection-troubleshooting.txt:7]: I: we have stopped using this feature
S: omit

File: source/connection-troubleshooting.txt [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+   :depth: 2
+   :class: singlecol
+
+This page offers potential solutions to issues you might encounter when
+using the {+driver-long+} to connect to a MongoDB deployment.
+
+.. note::
+
+   This page addresses only connection issues. If you encounter any other issues
+   with MongoDB, visit the following resources:
```
Comment (rustagir) [inline source/connection-troubleshooting.txt:21]:
```suggestion
   This page addresses only connection issues. If you encounter any other issues
   with MongoDB or the driver, visit the following resources:
```

File: source/connection-troubleshooting.txt [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+~~~~~~~~~~~~~~~~~~~~
+
+The {+driver-short+} can fail to connect to a MongoDB instance if
+the authorization is not configured correctly. In these cases, the driver raises
+an error message similar to one of the following messages:
+
+.. code-block:: none
+   :copyable: false
+
+   Command failed with error 18 (AuthenticationFailed): 'Authentication failed.' on server localhost:27017.
```
Comment (rustagir) [inline source/connection-troubleshooting.txt:78]: S: the next error block has line breaks. to reduce horizontal scroll, do you think you could insert a new line somewhere in this one too?

**Review by rustagir — CHANGES_REQUESTED** [review]
Great job addressing comments! Just a few more things

File: source/connection-troubleshooting.txt [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+Sometimes when you send a request through the driver to the server, the request
+times out. When this happens, you might receive an error message
+similar to the following message:
+
+.. code-block:: none
+   :copyable: false
+
+   timed out while checking out a connection from connection pool: context canceled
+
+If you receive one of these errors, try the following methods to resolve the
```
[the remark this answers is not in the document]
Comment (DBirtolo-mdb) [reply source/connection-troubleshooting.txt:218]: Oof! Good catch!

File: source/connection-troubleshooting.txt [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+issue.
+
+Set Timeout
```
[the remark this answers is not in the document]
Comment (DBirtolo-mdb) [reply source/connection-troubleshooting.txt:221]: I originally had this stylized as Set `Timeout` and then thought it looked weird. I'll put that formatting back in for clarity. But this is talking about setting the `Timeout` parameter.

File: source/connection-troubleshooting.txt [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+
+.. code-block:: none
+   :copyable: false
+
+   timed out while checking out a connection from connection pool: context canceled
+
+If you receive this error, try the following methods to resolve the
+issue.
+
+Set ``Timeout`` Option
```
Comment (rustagir) [inline source/connection-troubleshooting.txt:220]: S: remove monospace per the [style guide](https://www.mongodb.com/docs/meta/style-guide/style/titles-and-headings/#:~:text=Don%27t%20apply%20font%20treatments%20(bold%2C%20italics%2C%20or%20monospace)%20to%20text%20in%20a%20title%20or%20heading.)

File: source/faq.txt [hunk 8]
```diff
@@ -20,6 +20,13 @@ This page contains frequently asked questions and their corresponding answers.
    see the :ref:`golang-issues-and-help` page for next steps and more
    resources.
 
+Why Am I Getting Errors Connecting to MongoDB?
```
Comment (rustagir) [inline source/faq.txt:23]:
```suggestion
Why Am I Getting Errors While Connecting to MongoDB?
```

**Review by rustagir — APPROVED** [review]
lgtm - left some non blocking comments

**Pull request merged**